### PR TITLE
Tickets/dm 38716

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: c_src/connect_q330.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
+
+#ifndef stdbool_h
```
Comment (pkubanek) [inline c_src/connect_q330.c:2]: You should not need that, as all the headers provide their guards. Just:

#include <stdbool.h>
#include <stdio.h>
...

If that was in an example, whoever wrote that doesn't understand how this is handled with include guards.

File: c_src/q330.h [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+tminiseed get_miniseed();
+tonesec get_onesec();
+tstate get_state();
+void q330_create_context();
+void q330_unregistered_ping();
+void q330_register();
+void q330_change_state(enum tlibstate new_state);
+void q330_destroy_context();
+
+#endif
```
Comment (pkubanek) [inline c_src/q330.h:118]: As a good practice, I always add to a long #if .. #endif a comment indicating what ends. In this case:

#endif // !q330_h

with ! being for notdef.

File: c_src/connect_q330.c (hunk 1 above)
Comment (wvreeven) [reply c_src/connect_q330.c:2]: Fixed, thanks.

File: c_src/connect_q330.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+void q330_state_callback() {
+    tstate state = get_state();
+    printf("get_state: state_type='%d' station_name='%s' subtype='%d' info='%d' state_name='%s'.\n",
+           state.state_type, state.station_name, state.subtype, state.info, state.state_name);
+}
+
+int main() {
+    int short_sleep_time = 1;
+    int long_sleep_time = 25;
+    uint64_t serial_id = strtoll("0x01000018753C8C49", NULL, 16);
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply c_src/connect_q330.c]: Thanks, change pushed.

**Review by pkubanek — APPROVED** [review]
Great work to get ESS data working. Thanks.

**Pull request merged**